emma-lib/src/main/scala/org/emmalanguage/lib/ml/classification/naiveBayes.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright © 2017 TU Berlin (emma@dima.tu-berlin.de)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.emmalanguage
+package lib.ml.classification
+
+import api._
+import api.Meta.Projections._
+import lib.linalg._
+import lib.ml._
+
+@emma.lib
+object naiveBayes {
+
+  type ModelType = ModelType.Value
+  case class Model[L](label: L, pi: Double, theta: DVector)
+
+  def apply[ID: Meta, L: Meta](D: Int,
+    lambda: Double, modelType: ModelType // hyper-parameters
+  )(
+    data: DataBag[LDPoint[ID, L]] // data-parameters
+  ): DataBag[Model[L]] = {
+
+    val Sum = alg.Fold[LDPoint[ID, L], DVector](
```

Comment:
It might make sense to add this as an algebra to our collection in `lib.stats` or maybe use the one that's already defined for `DVectors` once we merge that branch.